File: remote/src/main/java/com/kelsos/mbrc/presenters/GenreArtistsPresenter.java
package com.kelsos.mbrc.presenters;

import com.kelsos.mbrc.annotations.Queue;
import com.kelsos.mbrc.domain.Artist;
import com.kelsos.mbrc.ui.views.GenreArtistView;

public interface GenreArtistsPresenter {
  void bind(GenreArtistView view);

  void onPause();

  void onResume();

  void load(long genreId);

  void queue(@Queue.Action String action, Artist artist);
}

File: app/src/main/kotlin/com/kelsos/mbrc/presenters/GenreArtistsPresenter.kt
package com.kelsos.mbrc.presenters

import com.kelsos.mbrc.annotations.Queue
import com.kelsos.mbrc.domain.Artist
import com.kelsos.mbrc.ui.views.GenreArtistView

interface GenreArtistsPresenter {
  fun bind(view: GenreArtistView)

  fun onPause()

  fun onResume()

  fun load(genreId: Long)

  fun queue(@Queue.Action action: String, artist: Artist)
}
